### PR TITLE
build: Add macOS_Catalina.md document

### PR DESCRIPTION
**Pull request by cclauss**
Document the lessons learned from #1779, #1861, #1927 and elsewhere to
enable users to consistently install `node-gyp` on macOS Catalina (v10.15).

@AshCripps  Your review please.

##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: macOS_Catalina.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+3. `softwareupdate -l`  # No listing is a good sign.
+    * If Xcode or Tools upgrades are listed, use "Software Upgrade" to install them.
+4. `xcode-select -version`  # Should return `xcode-select version 2370` or later.
+5. `xcode-select -print-path`  # Should return `/Library/Developer/CommandLineTools`
+6. Try the [_acid test_ steps above](#The-acid-test) to see if your Mac is ready.
+7. If the _acid test_ does _not_ pass then...
+8. `sudo xcode-select --reset`  # Enter root password.  No output is normal.
+9. Repeat step 5 above.  Is the path different this time?  Repeat the _acid test_.
+
+### I did all that and the acid test still does not pass :-(
```
Comment (GregBrimble) [inline macOS_Catalina.md]: This worked for me! Thank you :)

Comment (cclauss) [reply macOS_Catalina.md]: Could I please get your review on this PR?

**Comment by rvagg** [comment]
Two things I wouldn't mind seeing: a link from the README pointing out its existence (otherwise this will be ignored) and a TTL at the top of the doc, maybe something like this in italics:

> _This document specifically refers to upgrades from previous versions of macOS to Catalina (10.15). It should be removed from the source repository when Catalina ceases to be the latest macOS version or updated to deal with challenges involved in upgrades to the next macOS version._

An obvious signal for whoever's running this project in a ~year. Otherwise it's pure cruft making a mess in the root directory.

File: macOS_Catalina.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+1. `sudo rm -rf $(xcode-select -print-path)`  # Enter root password.  No output is normal.
```
Comment (peterquentin) [inline macOS_Catalina.md]: This actually worked for me, thnx!

**Comment by gengjiawen** [comment]
Maybe put in directory `troubleshooting` in case we have other similar problem ?

File: macOS_Catalina.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+8. Try the [_acid test_ steps above](#The-acid-test) to see if your Mac is ready.
+9. If the _acid test_ does _not_ pass then...
+10. `sudo xcode-select --reset`  # Enter root password.  No output is normal.
+11. Repeat step 7 above.  Is the path different this time?  Repeat the _acid test_.
+
+### Installing `node-gyp` using the Xcode Command Line Tools
+1. If the _acid test_ has not succeeded, then try `xcode-select --install`
+2. Wait until the install process is _complete_.
+3. `softwareupdate -l`  # No listing is a good sign.
+    * If Xcode or Tools upgrades are listed, use "Software Upgrade" to install them.
```
Comment (morinted) [inline macOS_Catalina.md]: I believe this is a small mistake. (repeated once in the previous section)

```suggestion
    * If Xcode or Tools upgrades are listed, use "Software Update" to install them.
```

File: macOS_Catalina.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    * If `com.apple.pkg.CLTools_Executables` is listed then try:
+2. `/usr/sbin/pkgutil --pkg-info com.apple.pkg.CLTools_Executables`
+    * If `version: 11.0.0` or later is listed then _you are done_!  Your Mac should be ready to install `node-gyp`.  Doing `clang -v` should show `Apple clang version 11.0.0` or later.
+
+As you go through the remainder of this document, at anytime you can try these `acid test` commands. If they pass then your Mac should be ready to install `node-gyp`.
+
+### Two roads
+There are two main ways to install `node-gyp` on macOS:
+1. With the full Xcode (~7.6 GB download) from the `App Store` app.
+2. With the _much_ smaller Xcode Command Line Tools via `xcode-select -install`
```
Comment (TheSimoms) [inline macOS_Catalina.md]: There's a "-" missing in "xcode-select -install"

**Comment by rvagg** [comment]
fixed, squashed and merged in 8ec2e681d536

**Pull request closed without merging**